core/lib/spree/app_configuration.rb

```diff
@@ -594,8 +588,13 @@ def stock
       @stock_configuration ||= Spree::Core::StockConfiguration.new
     end
 
+    # Allows providing your own promotion configuration instance
+    # @!attribute [rw] promotions
+    # @return [Spree::Core::PromotionConfiguration] an object that conforms to the API of
+    #   the standard promotion configuration class Spree::Core::PromotionConfiguration.
+    attr_writer :promotions
     def promotions
-      @promotion_configuration ||= promotion_configuration_class.new
```

Comment:
Maybe it goes beyond this PR, but are we sure we want to use `promotions` as the interface for a `PromotionConfiguration` object? I have the feeling something is off here, and this could be a good opportunity to refactor, what do you think?

Reply:
We're already in the context of a configuration object, and I feel that `Spree::Config.promotion_configuration` is a mouthful and doesn't provide any more information than `Spree::Config.promotions`. Especially because some of the methods in the configuration object have more `promotion`: `Spree::Config.promotion_configuration.promotions_adjuster_class.new(order)`, for example, has more duplication than I would wish for (I'm even thinking of shortening that before the release to just `adjuster_class`. 

So ideally, we'd have something like `Spree::Config.promotions.adjuster_class.new(order)`. IMO, shorter is better. 

However, I'm not willing to make this a hill I die on. If y'all think more verbosity is better, I'm all ears.

Reply:
Explained like this, it makes total sense. I was just missing some context. Going to approve this.